cmd/juju/commands/debuglog.go

```diff
@@ -213,36 +213,49 @@ func (c *debugLogCommand) processEntities(isCAAS bool, entities []string) []stri
 	if entities == nil {
 		return nil
 	}
-	result := make([]string, len(entities))
+	results := make([]string, len(entities))
 	for i, entity := range entities {
-		// A stringified unit or machine tag never match their "IsValid"
+		// Filter out valid tags early
+		if tag, err := names.ParseTag(entity); err == nil {
+			if tag.Kind() == names.UnitTagKind || tag.Kind() == names.MachineTagKind || tag.Kind() == names.ApplicationTagKind {
+				results[i] = entity
+				continue
+			}
+			logger.Warningf("Tag %s of kind %s is not a supported filter for debug-log", entity, tag.Kind())
```

Comment:
It is a little awkward. I decided against exiting out for 'safety' without really thinking through if it would actually keep anything safe. I'll exit out with an error I think

That will save a case of the weird mixed use of `ctx` and `logger` for logging in the client as well

Reply:
However, sometimes there can be some ambiguity whether our query is a already a tag or application name.

For instance, you could deploy ubuntu and name your application `charm-ubuntu` for instance